src/NetEscapades.EnumGenerators/SourceGenerationHelper.cs

```diff
@@ -962,4 +963,63 @@ public static string[] GetNames()
             .ToString();
         return (content, filename);
     }
+
+    private static IReadOnlyCollection<(string Key, EnumValueOption Value)> GetNamesOrderedByValue(EnumToGenerate enumToGenerate)
```

Comment:
I think it would make sense to make this `internal` and write some explicit unit tests to confirm it's ordering as expected. Right now, I'm not 100% what order we're _trying_ to output, and unit tests would make that clearer.